engine/src/main/java/org/terasology/internet/internal/InternetManagerImpl.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.internet.internal;
+
+import com.google.common.collect.Maps;
+import org.terasology.internet.InternetManager;
+import org.terasology.internet.TCPSocket;
+import org.terasology.naming.Name;
+import org.terasology.registry.DynamicInstanceProvider;
+
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.Map;
+import java.util.TreeSet;
+
+/**
+ * @author soniex2
+ */
+public class InternetManagerImpl implements InternetManager, DynamicInstanceProvider<InternetManager> {
```

Comment:
Please separate the provider from the implementation (i will add an additional comment about the per-module instance on the review)